### ISSUE TITLE
Bug or feature: "normalize" analyzes all channels without downmixing them at first

### ISSUE THREAD

**Issue by ronny332** [staxrip/staxrip#1452]
Hi there!

as the topic says, I'm unsure if it is by design, or a mistake:

when downmixing any source to stereo (that has an channel amount of not 2) the source gets analyzed before downmixing it.

In my eyes this doesn't make sense, it should be downmixed at first, than the volumedection of ffmpeg should run.

I haven't found any similar question what makes me wonder if I'm doing anything wrong.

Thank you!

**Comment by Dendraspis** [comment]
Don't forget, that StaxRip is over 20 years old and procedures developed or were added.

Under normal circumstances you won't get/hear any big differences and normal procedures won't let you get into situations where it could become an issue. But technically it would be better to normalize at the end, yes.

I have made a test using a 5.1 source and converted it to stereo with modified channel volumes to make the center a bit louder for better voice. The resulting FLAC was then checked again and its max volume reported by `ffmpeg` was:  `0.0dB`

Maybe I will try to change it in the future, but currently I don't see the result getting so off, that it could cause issues like distortion or so.

**Comment by ronny332** [comment]
Hi @Dendraspis . Thanks for your reply.

My tests show a different behavior. Every 5.1 gets downmxed in the one possible every way, every channel has to have the volume to avoid clipping if all channels have a maximum volume on its own.
When you try to get the maximum volume for a channel of the 5.1 streams and downmix the channels afterwards to 2.0 they are very very quiet. I added +9dB and there was still a lot of room to 0dB.
But that's not a problem of Staxrip, that's the normal behavior.

What I see as issue is just the order in which the steps occur. The volumedetection should be done after downmixing, not before.

I worked around the issue by downmixing my files to 2.0 pcm_s24le with ffmpeg and put the files to staxrip aftwards. Works not as desired :-). Everything right at 0dB.

**Comment by Dendraspis** [comment]
Yes, the voice gets sort of silent, but I was talking about the technical aspect.

`v2.43` will have an option included to downmix differently to Stereo, meaning adjusted channel volumes. That should also make a difference.

> What I see as issue is just the order in which the steps occur. The volumedetection should be done after downmixing, not before.

I agree, even when the difference is not noticable under normal conditions when using StaxRip.
